### ISSUE TITLE
New cdo=1.9.7.1 that is mpi-agnostic depends on proj4>=6.1.1 that is causing massive conflicts

### ISSUE THREAD

**Issue by valeriupredoi** [conda-forge/cdo-feedstock#65]
Issue: proj4>6.1.1 as dependency for cdo=1.9.7.1 (mpi agnostic, `=h7d1db5e_1`) is causing a lot of conflicts since a lot of packages from conda-forge are limiting proj4 versions to <6.1.1

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
(base)
```
No point adding the conda list output since I am using the `(base)` from conda=4.7.11 and trying to create a TEST environment like this:
```
---
name: cdoenv
channels:
  - conda-forge

dependencies:
  - libgdal
  - cdo=1.9.7.1=h7d1db5e_1
```

That fails since 

```
Package proj4 conflicts for:
libgdal -> proj4[version='5.2.0.*|>=5.2.0,<5.2.1.0a0|>=6.1.0,<6.1.1.0a0']
cdo==1.9.7.1=h7d1db5e_1 -> proj4[version='>=6.1.1,<6.1.2.0a0']
Note that strict channel priority may have removed packages required for satisfiability.
```

Is the new `cdo` not compatible with older `proj4` versions like `5.2.0`?

**Comment by valeriupredoi** [comment]
@ocefpaf was very helpful in the very recent past so I am tagging you :beer: Cheers in advance!

**Comment by Try2Code** [comment]
do you really need the changes from 6.1.0 to 6.1.1 @valeriupredoi ?

**Comment by valeriupredoi** [comment]
> do you really need the changes from 6.1.0 to 6.1.1 @valeriupredoi ?

what do you mean? I need to use the latest `cdo` that is not asking for `fftw` that depends on `openmpi` in an environment with stuff that use `libgdal` and other packages that, in turn, don't allow for `proj4=6.1.1` but need `6.1.0` or lower (well 6.1.0 is fine at the mo)

**Comment by Try2Code** [comment]
The cdo build could be downgraded to proj=6.1.0, but then other people with a similar environment like u (but relying on 6.1.1) will have similar problems.

what about libgdal - could it be rebuild with proj=6.1.1 to solve the issue?

**Comment by ocefpaf** [comment]
Note that conda-forge has a central pinning and, as packages evolve, it gets harder and harder to build for old packages.

If you want a version with an older proj you can try to send a PR with a pin to the version that you need. Remember to bump the build number. Also, that pin would be a one-time build b/c we need to remove it for future versions to get the stack consistent with the rest of `conda-forge`.

**Comment by valeriupredoi** [comment]
our env fails to solve due to `proj4>=6.1.1` needed as dependency for `cdo` and that happens due to a bunch of packages not playing well with `6.1.1` just yet, I used `libgdal` as a minimal example, see what conda says when I try build our environment:
```
Package proj4 conflicts for:
ncl[version='>=6.5.0'] -> libgdal[version='>=2.4.0,<2.5.0a0'] -> proj4=5.2.0
cdo==1.9.7.1=h7d1db5e_1 -> proj4[version='>=6.1.1,<6.1.2.0a0']
iris[version='>=2.2'] -> cartopy[version='>=0.14'] -> owslib -> pyproj -> proj4[version='6.0.0.*|>=4.9.3,<4.9.4.0a0|>=5.2,<6|>=6.0.0,<6.0.1.0a0|>=6.1.1,<6.1.2.0a0']
ncl[version='>=6.5.0'] -> proj4[version='>=5.2.0,<5.2.1.0a0|>=6.1.0,<6.1.1.0a0']
iris[version='>=2.2'] -> cartopy[version='>=0.14'] -> proj4[version='5.2.0.*|>=5.2.0,<5.2.1.0a0|>=6.1.0,<6.1.1.0a0']
proj4=6.1.1
esmvalcore[version='>=2.0.0b0,<2.1'] -> iris[version='>=2.2'] -> cartopy[version='>=0.14'] -> proj4[version='5.2.0.*|>=5.2.0,<5.2.1.0a0|>=6.1.0,<6.1.1.0a0']
esmvalcore[version='>=2.0.0b0,<2.1'] -> iris[version='>=2.2'] -> cartopy[version='>=0.14'] -> owslib -> pyproj -> proj4[version='6.0.0.*|>=4.9.3,<4.9.4.0a0|>=5.2,<6|>=6.0.0,<6.0.1.0a0|>=6.1.1,<6.1.2.0a0']
Note that strict channel priority may have removed packages required for satisfiability.
```

`proj4=6.1.1` is fairly new (about a month old) and I reckon it's probably natural to offer support for something that's at least one release older, those people you mention most probably don't need a fairly complex environment because if they did, they'd run in he same issue as us, dependencies that have not yet updated to allow for the latest `proj4`. The question is - could you guys relax the dep requirements for proj4 to something like `"proj4 >=6.1.0,<6.1.2.0a0"` that will allow for 6.1.1 as well without having to rebuild the whole of cdo?

**Comment by valeriupredoi** [comment]
@ocefpaf - fair and square but `proj4=6.1.0` is surely not old - they have just released a new hash for `5.2.0` a day ago and obviously keep maintaining it, 6.1.1 is a month old (as I mentioned above) and surely you don't expect all the packages from conda-forge to have their dependency versions aligned in one month :grin:

**Comment by valeriupredoi** [comment]
things have started to improve and now we can actually build a working environment with the latest mpi-agnostic `cdo` and `proj4=6.1.1`, the only major show stopper at the mo is `ncl` as a dep, but hopefully this will be sorted soon too, so am gonna close this issue and thank the guys who commented :beer:
